napari/utils/colormaps/colormap_utils.py

```diff
@@ -29,20 +23,70 @@
 ]
 
 
-primary_color_names = ['red', 'green', 'blue', 'cyan', 'magenta', 'yellow']
-primary_colors = np.array(
-    [(1, 0, 0), (0, 1, 0), (0, 0, 1), (0, 1, 1), (1, 0, 1), (1, 1, 0)],
-    dtype=float,
+matplotlib_colormaps = _MATPLOTLIB_COLORMAP_NAMES = OrderedDict(
+    viridis=trans._('viridis'),
+    magma=trans._('magma'),
+    inferno=trans._('inferno'),
+    plasma=trans._('plasma'),
+    gray=trans._('gray'),
+    gray_r=trans._('gray r'),
+    hsv=trans._('hsv'),
+    turbo=trans._('turbo'),
+    twilight=trans._('twilight'),
+    twilight_shifted=trans._('twilight shifted'),
+    gist_earth=trans._('gist earth'),
+    PiYG=trans._('PiYG'),
+)
```

Comment:
Is that really translatable ? They seem like proper name or at least needs some context, especially with things like `gray r` where r is for revered...

Reply:
>Is that really translatable ?

There may be a better mor ereadable name, in english as well I mean.... so... maybe :) ?

Reply:
Overall with this PR I kind of agree with @Carreau's sentiment that maybe we maybe don't want to localize these at all. A lot of these are documented in the matplotlib docs, which are (to my knowledge?) not translated. Many of these at a minimum need context, e.g. in "gist earth", "gist" does not have the common English meaning, but rather is an acronym/software name, see https://www.ctcms.nist.gov/~langer/oof2man/RegisteredClass-GistEarth.html. Another idea would be to only translate some ('autumn', 'blue', 'bop blue', etc), and for those that are translated, somehow show the original name next to it. (e.g. "azul (blue)"). What do you think @goanpeca?

Reply:
What's the Spanish for "colormap" anyway, @goanpeca? 😂

Reply:
Please, do not translate these names. It will produce much more problems than solve.

Reply:
>What do you think @goanpeca?

It is much easier to maintain adding all or removing all, in this case some make sense to translate some other might not, but overall this is more of a decision of the translator. They can choose to leave the same string, but we still leave the possibility open.

`mapa de colores` @jni?

Reply:
> Please, do not translate these names. It will produce much more problems than solve.

@Czaki can you please explain what specific problems you are concerned about. Generic negative comments like this are not helpful as we don't know what you are worried about, and we're trying to put in a lot of effort here to ensure that napari is most broadly accessible and easy to use

Reply:
I see few big problems:

1. Different names in API and UI. 
2. When you read the original tutorial and use a translated version of napari part of the translations are simple or you see the position of elements on screenshots. When the name of colormap is provided you need to guess the colormap name. 
3. If someone has started his journey in scientific python from napari then he still needs to learn original names as other libraries used them. 

Additionally, napari/product-heuristics-2020#14 was created before #1862 was created and merged. Now the user has a preview of how colormaps look and do not need to know it based on names.

As it is mentioned at Europe/Asia meetings the food example of too many translations/localizations is Excel where every function has translated name and function argument separation depends on local settings. It simplifies starting Excel but create enormous pain if you need to cooperate with a foreign person or have two computer with different settings (for example your private - native and work one in English)

Reply:
For different names in the API/ UI - we've already started down that road with things like the `layer.blending = 'additive'` example that i mention below (you would now see 'aditivo' in the UI, but the API would be 'additive'). I really don't see a way around that, it is always going to be `layer.blending` and not `layer.mezclando` (i.e. attribute names that appear in the UI are being translated). I guess we could have the option to enable/ disable translations on the strings that are settable, but I wonder how many people would want to see `mezclando: additive` in the UI (mixing spanish and english) instead of either seeing `mezclando: aditivo` (all spanish) or `blending: additive` (all english). I also think in time the vast majority of users will only interact with the UI and not the API (there are just many more people doing scientific imaging than know how to or are interested in coding python) and some of those may want to opt into translation. For users that are heavily using the API they might want to turn translation off.

> When the name of colormap is provided you need to guess the colormap name.

We could provide the untranslated name here too, so `rojo (red)`, but I still don't get why a colormap is so different than a blending mode. Maybe all the strings that are settable should retain their original, so `mezclando: aditivo (additive)` to make the link to the API clear, though this could get cluttered and leave us with a lot of `additive (additive)` (though i guess we could check to see if the strings were different before doing that). Maybe this would solve peoples concerns (thoughts on that idea @goanpeca ?)

> too many translations/localizations is Excel where every function has translated name and function argument separation depends on local settings. It simplifies starting Excel but create enormous pain if you need to cooperate with a foreign person or have two computer with different settings (for example your private - native and work one in English)

excel has been a remarkably successful program, and i think we're trying to get to that ease of someone starting. We're also making it easy to change translation in our settings and preferences so people can turn it on and off as they wish.

Reply:
Priding both names may create really long strings and make the interface less readable.

Reply:
@sofroniewn I don't think @Czaki wanted to imply that colormaps are different from blending modes — he probably prefers for neither to be translated. (Side note: I hope "mezclando" is just Google Translate speaking 😂 I think it should be "mezcla" since blending is a noun here. Possibly that hints to another `_p()` requirement.)

The move from names to display names in the UI does slightly hurt the API usability in my opinion, as now there needs to be a mental translation from UI to API if someone wants to translate their UI fiddling into code. I would suggest using the parens but *only* in the cases where `display_name != name`.

I also think Excel has a much bigger audience than napari, the latter of which is primarily for *scientists*, which de-facto work in English already. And re: Python vs UI users, it's a fair point that there will be more UI users, but I really think that we should strive to not worsen the Python experience. We've always said (it's in my blog post!) that we also want napari to be a "back door" to getting people to learn Python (e.g. through script generation), and the bigger the disconnect between the UI and the API, the more difficult that is.

It's a tension I know, and accessibility is important. But I just want to make sure we proceed with caution here.

Reply:
Proceeding with caution is fine, i think making `colormap._display_name` private will help ensure that there is no confusion between the API and the UI for people writing scripts. Otherwise the translations are opt in, and we'll make easy to turn on and off in case there is confusion. I really think it's the people that we're not hearing from right now that will benefit from this functionality